api/http_client.go

```diff
@@ -25,8 +28,25 @@ func responseReaderAndStatus(url string) (io.Reader, int, error) {
 	if err != nil {
 		return nil, 0, err
 	}
-
 	status := resp.StatusCode
+
+	// Let's not let Canonical's server troubles ruin our day. Retry a few times on 5xx.
+	for retryCount := 1; retryCount < 6; retryCount++ {
+		if status < 500 || status > 599 {
+			break
+		} else {
```

Comment:
Golang convention would be to omit the `else`. There may also be a way to put the first `http.DefaultClient.Do(req)` into the loop so that there aren't two places in the code that make the same request.

Reply:
Thanks much for the review.

> There may also be a way to put the first 'http.DefaultClient.Do(req)' into the loop so that there aren't two places in the code that make the same request.

Yeah, I don't at all like that I duplicated the call and error-handling. I thought about it for a couple of minutes when I was writing the code, and then decided that it would take me too long to figure out how to make a safe `*Request` object instance for the initial value outside the loop ("Surely the go-equivalent of `NULL` would be unsafe, right?", I thought to myself.).